packages/eas-cli/src/commands/build/index.ts

```diff
@@ -117,20 +133,13 @@ export default class Build extends EasCommand {
       Errors.error('--json is allowed only when building in non-interactive mode', { exit: 1 });
     }
 
-    const requestedPlatform = await selectRequestedPlatformAsync(flags.platform);
-    if (flags.local) {
-      if (flags['auto-submit'] || flags['auto-submit-with-profile'] !== undefined) {
-        // TODO: implement this
-        Errors.error('Auto-submits are not yet supported when building locally', { exit: 1 });
-      }
-
-      if (requestedPlatform === RequestedPlatform.All) {
-        Errors.error('Builds for multiple platforms are not supported with flag --local', {
-          exit: 1,
-        });
-      } else if (process.platform !== 'darwin' && requestedPlatform === RequestedPlatform.Ios) {
-        Errors.error('Unsupported platform, macOS is required to build apps for iOS', { exit: 1 });
-      }
+    const requestedPlatform =
+      flags.platform &&
+      Object.values(RequestedPlatform).includes(flags.platform.toLowerCase() as RequestedPlatform)
+        ? (flags.platform.toLowerCase() as RequestedPlatform)
+        : undefined;
+    if (requestedPlatform) {
+      Build.sanitizePlatform(flags, requestedPlatform);
```

Comment:
This is not necessary here, that validation can happen inside `sanitizePlatform` function even if the platform is passed as a param.

Also sanitizePlatform currently only validates flags for local builds, and in that case we probably do not need to show outages.